src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -180,11 +193,79 @@ public static class PrereleaseCompatibilityTransformer
             return transformedSarifLog;
         }
 
-        private static bool ApplyRtm0Changes(JObject sarifLog)
+        private static bool ApplyRtm1Changes(JObject sarifLog)
         {
             UpdateSarifLogVersionAndSchema(sarifLog);
             bool modifiedLog = false;
 
+            if (sarifLog["runs"] is JArray runs)
+            {
+                foreach (JObject run in runs)
+                {
+                    // https://github.com/oasis-tcs/sarif-spec/issues/414
+                    modifiedLog |= ConvertResultLogicalLocationToArray(run);
+                }
+            }
+
+            return modifiedLog;
+        }
+
+        private static bool ConvertResultLogicalLocationToArray(JObject sarifLog)
+        {
+            // note: intentionally leaving inlineExternalProperties refs - since no one is using them yet.
+            string[] resultPathsToUpdate =
+            {
+                "results[].locations[]",
```

Comment:
>results[].locations [](start = 17, length = 19)

doesn't this data exactly correspond to another traversal we did of all locations in the format? we should DRY the data out if so